src/Elcodi/Admin/UserBundle/Security/OneTimeLoginAuthenticator.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/*
+ * This file is part of the Elcodi package.
+ *
+ * Copyright (c) 2014 Elcodi.com
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * Feel free to edit as you please, and have fun.
+ *
+ * @author Marc Morera <yuhu@mmoreram.com>
+ * @author Aldo Chiecchia <zimage@tiscali.it>
+ */
+
+namespace Elcodi\Admin\UserBundle\Security;
+
+use Doctrine\Common\Persistence\ObjectManager;
+use Doctrine\Common\Persistence\ObjectRepository;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Security\Core\Authentication\SimplePreAuthenticatorInterface;
+use Symfony\Component\Security\Core\Authentication\Token\PreAuthenticatedToken;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\Exception\BadCredentialsException;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+
+/**
+ * Class OneTimeLoginAuthenticator responsible of checking the request for a login hash and login the user in this case.
+ */
+class OneTimeLoginAuthenticator implements SimplePreAuthenticatorInterface
+{
+    /**
+     * The Admin user repository object.
+     *
+     * @var ObjectRepository
+     */
+    protected $adminUserObjectRepository;
+
+    /**
+     * The admin user manager object.
+     *
+     * @var ObjectManager
+     */
+    private $adminUserManager;
+
+    /**
+     * Generates a new OneTimeLoginAuthenticator injecting it's dependencies.
+     *
+     * @param ObjectManager    $adminUserManager          The admin user manager object.
+     * @param ObjectRepository $adminUserObjectRepository The Admin user repository object.
+     */
+    public function __construct(ObjectManager $adminUserManager, ObjectRepository $adminUserObjectRepository)
```

Comment:
When the line in constructor is that long, we are used to adding a new line per each parameter